pulsar-broker/src/test/java/org/apache/pulsar/schema/SchemaTest.java

```diff
@@ -708,7 +709,7 @@ public void testNullKeyValueProperty() throws PulsarAdminException, PulsarClient
         map.put(null, "value"); // null key is not allowed for JSON, it's only for test here
 
         // leave INT32 instance unchanged
-        final Schema<Integer> integerSchema = Schema.INT32.clone();
+        final Schema<Integer> integerSchema = DefaultImplementation.getDefaultImplementation().newIntSchema();
```

Comment:
This is not good.
it means that users that use the public API Schema.INT32 and use clone() see surprises.

I believe that the problem is that clone() is not working properly 

Schema are stateful objects in Pulsar and clone() is very important

Reply:
if you don't want to open the Pandora box of clone() then please use Schema.JSON() that is guaranteed to always return a new instance

Reply:
I don't believe this solution works. `SCHEMA_INFO` is static, so any modification of it in one test will affect other tests. This is a design flaw.

A larger question for me is why we're letting users modify the `SCHEMA_INFO` object. It is static, so cloning isn't going to do anything.

Reply:
@michaeljmarshall good catch! you're right
I think this issue can be just fixed as @eolivelli suggested, using a JSON schema. The type of the schema is not relevant for test itself.

I agree that the `clone()` method return `this` because probably the intention was to keep it immutable. 
Moving the getter to return an immutable map may be the right solution, even if it can be considered breaking change because can cause runtime error while upgrading the client

Reply:
Great point. In order to get this test passing, we can avoid the mutability issues for `Schema.INT32`.